mesonbuild/interpreter/interpreter.py

```diff
@@ -1534,6 +1534,10 @@ def add_find_program_override(self, name: str, exe: T.Union[build.Executable, Ex
             raise InterpreterException(f'Tried to override finding of executable "{name}" which has already been found.')
         if name in self.build.find_overrides:
             raise InterpreterException(f'Tried to override executable "{name}" which has already been overridden.')
+        if isinstance(exe, build.Executable) and exe.for_machine == MachineChoice.HOST and \
+           not self.environment.can_run_host_binaries():
+            FeatureDeprecated.single_use('Override program with an executable that cannot be run.',
+                                         '0.64.0', self.subproject, location=self.current_node)
```

Comment:
I think this should just be a regular mlog.warning, because it's always a problem. There's nothing about 0.64.0 specifically that makes it problematic.

Reply:
Initially I wanted to make it fatal error, but it risk to break cross compile of some projects. Didn't we have a way to have a deprecation that always print the warning, and becomes fatal when targetting that Meson version?

Anyway, mlog.warning() is good enough, made that change.